src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,89 @@
 package mate.academy.lib;
 
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.exception.InizializationException;
+
 public class Injector {
+    public static final String CLASS_END_NAME = ".class";
+    public static final String PACKAGE_NAME = "mate.academy.service.impl";
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field declaredField : declaredFields) {
+            if (declaredField.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(declaredField.getType());
+
+                clazzImplementationInstance = getOrCreateNewInstance(clazz);
+
+                declaredField.setAccessible(true);
+                try {
+                    declaredField.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new InizializationException("Can't initialize field value. "
+                            + "class: " + clazz.getName() + ".Field: "
+                            + declaredField.getName(), e);
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = getOrCreateNewInstance(clazz);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object getOrCreateNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("can't create new instance of: " + clazz.getName(), e);
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
```

Comment:
still missing check if `@Component` is present on the impl class that we have found, this is part of the requirements, so lets add it